python_ta/__init__.py

```diff
@@ -92,8 +95,26 @@ def _check(module_name='', level='all', local_config='', output=None):
                 linter.check(file_py)  # Lint !
                 current_reporter.print_messages(level)
                 current_reporter.reset_messages()  # Clear lists for any next file.
+                f_paths.append(file_py)  # Appending paths for (potential) upload
                 print('[INFO] File: {} was checked using the configuration file: {}'.format(
                     file_py, linter.config_file))
+
+            if linter.config.pyta_error_permission or linter.config.pyta_file_permission:
+                errs = [msg for msg in current_reporter.messages_by_file if msg.filename in f_paths]
+                if linter.config.pyta_file_permission:
+                    upload_to_server(errors=errs,
+                                     config=linter.config.__dict__,
+                                     default=reset_linter().config.__dict__,
```

Comment:
This is better but still not the best approach. You shouldn't pass in an entire dict, that's not necessary to check for whether the default configuration is being used. Instead, try checking the attributes of `linter.config` to see if there's (e.g.) a path the specifies the configuration file that was used.